### DISCUSSION TITLE
Read data after plug down/up SENSOR

### DISCUSSION THREAD

**Discussion by SERIDJ** [UncleRus/esp-idf-lib#278] [General]
hi, 
  
i want to know if its possible to read data from any sensor without reboot the MCU (esp32) after disconnect and reconnect the sensor.

for example i try to call xxx_init_desc focntion . **(mcp23x17_init_desc(&dev, 0, MCP23X17_ADDR_BASE, SDA_GPIO, SCL_GPIO));** after free mux  **xxx_free_desc**

but it doesn't work with all sensor  ,

**Comment by UncleRus** [comment]
This usually does not require any special actions. However, it is possible that the particular I2C device needs to be reconfigured again after it loses power.
